include/sys/spa.h

```diff
@@ -1182,6 +1182,11 @@ extern int spa_wait_tag(const char *name, zpool_wait_activity_t activity,
 extern void spa_notify_waiters(spa_t *spa);
 extern void spa_wake_waiters(spa_t *spa);
 
+extern void spa_create_os(spa_t *spa);
+extern void spa_export_os(spa_t *spa);
+extern void spa_activate_os(spa_t *spa);
+extern void spa_deactivate_os(spa_t *spa);
```

Comment:
nit: let's move these private `_os` hooks in to `include/sys/spa_impl.h` with the others.

Reply:
OK in `spa_impl.h` means adding that include to `kernel.c` or, leaving it in `spa.h` and no change to `kernel.c`. Which do you prefer? I didn't find any other _os prototypes (in spa*)